paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -156,11 +243,23 @@ class MatMulFactory {
 
   void Execute() {
     dnnl::stream stream(engine_);
-    matmul_prim_.execute(stream, {
-                                     {MKLDNN_ARG_SRC, x_mem_},
-                                     {MKLDNN_ARG_WEIGHTS, y_mem_},
-                                     {MKLDNN_ARG_DST, out_mem_},
-                                 });
+
+    void* x_ptr = x_mem_.get_data_handle();
+    void* y_ptr = y_mem_.get_data_handle();
+    void* out_ptr = out_mem_.get_data_handle();
+    for (int64_t i = 0; i < batch_size_; i++) {
```

Comment:
You may want to save `batch_size_` to a local variable to avoid unnecessary cache misses due to aliasing problem.
Similarly, you may want to compute the `<smth>_offset_`s here instead of storing them as class members.